server/modules/record/manager/_recordManager/nodeUpdateDependentManager.js

```diff
@@ -114,12 +114,16 @@ export const updateSelfAndDependentsDefaultValues = async (survey, record, node,
         R.unless(R.isNil, (value) => RecordExpressionValueConverter.toNodeValue(survey, record, nodeCtx, value))
       )(exprEval)
 
-      // 4. persist updated node value if changed, and return updated node
-
+      // 4. persist updated node value if changed and new value is not empty, and return updated node
       if (R.equals(oldValue, exprValue)) {
         return {}
       }
 
+      // if value is empty and node is Taxon leave empty, the user likes to clean the value
+      if (R.isEmpty(oldValue) && NodeDef.isTaxon(nodeDef)) {
+        return {}
+      }
```

Comment:
I just realised this happens to date, time, taxon, code, boolean, decimal, integer..
we should find a more consistent way to fix this 

Uploading Screen Recording 2021-03-03 at 05.42.32.mov…

Reply:
I had the solution for that on the prev commit, but I reduced the scope only for the case of the bug, 

I will make changes to avoid default values on edit.

Reply:
the idea with this new line was add some prop to the NodeDEfs like cleanable 

If (R.isEmpty(oldValue) && NodeDef.isCleanable(nodeDef)) 

and allow overwriting with default values on value clean only if the NodeDef is not cleanable

Reply:
I think it's simply that the default value should not be applied on the node users are currently editing.
the concept of cleanable doesn't sound very robust

Reply:
okey i will proceed in that way, avoid default on update